### ISSUE TITLE
Multiprocessing attempts to run .ipynb on process launch

### ISSUE THREAD

**Issue by mrahtz** [jupyterlab/jupyterlab#13970]
## Description

When I start a `multiprocessing` process, it appears to try and run the notebook `.ipynb` the process was started from directly with the interpreter. If the `.ipynb` JSON contains any keywords that are not valid Python such as `true` or `null`, then the process raises an exception.

## Reproduce

0. Create a test environment:

```
$ conda create -n jlab-test --override-channels --strict-channel-priority -c conda-forge -c nodefaults jupyterlab
```

1. Create `foolib.py` containing:

```python
def foo():
    print('foo')
```

2. Create `foo.ipynb`, containing the following code, followed by a collapsed Markdown header cell: (eg https://gist.github.com/mrahtz/9617f8ed3cb187c972d6f79670a1f5cb)

```python
import multiprocessing
import foolib
multiprocessing.Process(target=foolib.foo).start()
```

3. Run the cell. An exception is raised:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/matthew/miniforge3/envs/jlab-test/lib/python3.11/multiprocessing/spawn.py", line 120, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthew/miniforge3/envs/jlab-test/lib/python3.11/multiprocessing/spawn.py", line 129, in _main
    prepare(preparation_data)
  File "/Users/matthew/miniforge3/envs/jlab-test/lib/python3.11/multiprocessing/spawn.py", line 240, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/Users/matthew/miniforge3/envs/jlab-test/lib/python3.11/multiprocessing/spawn.py", line 291, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 291, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/matthew/foo.ipynb", line 29, in <module>
    "jp-MarkdownHeadingCollapsed": true,
                                   ^^^^
NameError: name 'true' is not defined. Did you mean: 'True'?
```

## Expected behavior

I actually have no clue what _is_ supposed to happen here: the code in [`multiprocessing/spawn.py`](https://github.com/python/cpython/blob/6ef6915d3530e844243893f91bf4bd702dfef570/Lib/multiprocessing/spawn.py#L237) seems pretty adamant that new processes created with the _spawn_ method should execute the `__main__` module, which in a notebook is the `.ipynb` file.

## Context

- Operating System and version: macOS Ventura 13.2
- Browser and version: Chrome 110.0.5481.77
- JupyterLab version: 3.6.1

This has only started happening for me since I did a fresh install of macOS a couple of days ago. In my previous installation I was also using a Miniforge installation, I _think_ on Python 3.10.8, though I'm not sure of the package versions.

<details><summary>Troubleshoot Output</summary>
<pre>
$PATH:
	/Users/matthew/miniforge3/envs/jlab-test/bin
	/Users/matthew/miniforge3/condabin
	/Users/matthew/Library/Python/3.9/bin
	/opt/homebrew/bin
	/usr/local/bin
	/System/Cryptexes/App/usr/bin
	/usr/bin
	/bin
	/usr/sbin
	/sbin

sys.path:
	/Users/matthew/miniforge3/envs/jlab-test/bin
	/Users/matthew/miniforge3/envs/jlab-test/lib/python311.zip
	/Users/matthew/miniforge3/envs/jlab-test/lib/python3.11
	/Users/matthew/miniforge3/envs/jlab-test/lib/python3.11/lib-dynload
	/Users/matthew/miniforge3/envs/jlab-test/lib/python3.11/site-packages

sys.executable:
	/Users/matthew/miniforge3/envs/jlab-test/bin/python

sys.version:
	3.11.0 | packaged by conda-forge | (main, Jan 14 2023, 12:26:40) [Clang 14.0.6 ]

platform.platform():
	macOS-13.2-arm64-arm-64bit

which -a jupyter:
	/Users/matthew/miniforge3/envs/jlab-test/bin/jupyter
	/Users/matthew/Library/Python/3.9/bin/jupyter

pip list:
	Package                       Version
	----------------------------- -----------
	aiofiles                      22.1.0
	aiosqlite                     0.18.0
	anyio                         3.6.2
	appnope                       0.1.3
	argon2-cffi                   21.3.0
	argon2-cffi-bindings          21.2.0
	asttokens                     2.2.1
	attrs                         22.2.0
	Babel                         2.11.0
	backcall                      0.2.0
	backports.functools-lru-cache 1.6.4
	beautifulsoup4                4.11.2
	bleach                        6.0.0
	brotlipy                      0.7.0
	certifi                       2022.12.7
	cffi                          1.15.1
	charset-normalizer            2.1.1
	comm                          0.1.2
	cryptography                  39.0.1
	debugpy                       1.6.6
	decorator                     5.1.1
	defusedxml                    0.7.1
	entrypoints                   0.4
	executing                     1.2.0
	fastjsonschema                2.16.2
	flit_core                     3.8.0
	idna                          3.4
	importlib-metadata            6.0.0
	importlib-resources           5.10.2
	ipykernel                     6.21.1
	ipython                       8.10.0
	ipython-genutils              0.2.0
	jedi                          0.18.2
	Jinja2                        3.1.2
	json5                         0.9.5
	jsonschema                    4.17.3
	jupyter_client                8.0.2
	jupyter_core                  5.2.0
	jupyter-events                0.5.0
	jupyter_server                2.2.1
	jupyter_server_fileid         0.6.0
	jupyter_server_terminals      0.4.4
	jupyter_server_ydoc           0.6.1
	jupyter-ydoc                  0.2.2
	jupyterlab                    3.6.1
	jupyterlab-pygments           0.2.2
	jupyterlab_server             2.19.0
	MarkupSafe                    2.1.2
	matplotlib-inline             0.1.6
	mistune                       2.0.5
	nbclassic                     0.5.1
	nbclient                      0.7.2
	nbconvert                     7.2.9
	nbformat                      5.7.3
	nest-asyncio                  1.5.6
	notebook                      6.5.2
	notebook_shim                 0.2.2
	packaging                     23.0
	pandocfilters                 1.5.0
	parso                         0.8.3
	pexpect                       4.8.0
	pickleshare                   0.7.5
	pip                           23.0
	pkgutil_resolve_name          1.3.10
	platformdirs                  3.0.0
	prometheus-client             0.16.0
	prompt-toolkit                3.0.36
	psutil                        5.9.4
	ptyprocess                    0.7.0
	pure-eval                     0.2.2
	pycparser                     2.21
	Pygments                      2.14.0
	pyOpenSSL                     23.0.0
	pyrsistent                    0.19.3
	PySocks                       1.7.1
	python-dateutil               2.8.2
	python-json-logger            2.0.4
	pytz                          2022.7.1
	PyYAML                        6.0
	pyzmq                         25.0.0
	requests                      2.28.2
	Send2Trash                    1.8.0
	setuptools                    67.1.0
	six                           1.16.0
	sniffio                       1.3.0
	soupsieve                     2.3.2.post1
	stack-data                    0.6.2
	terminado                     0.17.1
	tinycss2                      1.2.1
	tomli                         2.0.1
	tornado                       6.2
	traitlets                     5.9.0
	typing_extensions             4.4.0
	urllib3                       1.26.14
	wcwidth                       0.2.6
	webencodings                  0.5.1
	websocket-client              1.5.1
	wheel                         0.38.4
	y-py                          0.5.5
	ypy-websocket                 0.8.2
	zipp                          3.13.0

conda list:
	# packages in environment at /Users/matthew/miniforge3/envs/jlab-test:
	#
	# Name                    Version                   Build  Channel
	aiofiles                  22.1.0             pyhd8ed1ab_0    conda-forge
	aiosqlite                 0.18.0             pyhd8ed1ab_0    conda-forge
	anyio                     3.6.2              pyhd8ed1ab_0    conda-forge
	appnope                   0.1.3              pyhd8ed1ab_0    conda-forge
	argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
	argon2-cffi-bindings      21.2.0          py311he2be06e_3    conda-forge
	asttokens                 2.2.1              pyhd8ed1ab_0    conda-forge
	attrs                     22.2.0             pyh71513ae_0    conda-forge
	babel                     2.11.0             pyhd8ed1ab_0    conda-forge
	backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
	backports                 1.0                pyhd8ed1ab_3    conda-forge
	backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
	beautifulsoup4            4.11.2             pyha770c72_0    conda-forge
	bleach                    6.0.0              pyhd8ed1ab_0    conda-forge
	brotlipy                  0.7.0           py311he2be06e_1005    conda-forge
	bzip2                     1.0.8                h3422bc3_4    conda-forge
	ca-certificates           2022.12.7            h4653dfc_0    conda-forge
	certifi                   2022.12.7          pyhd8ed1ab_0    conda-forge
	cffi                      1.15.1          py311hae827db_3    conda-forge
	charset-normalizer        2.1.1              pyhd8ed1ab_0    conda-forge
	comm                      0.1.2              pyhd8ed1ab_0    conda-forge
	cryptography              39.0.1          py311h507f6e9_0    conda-forge
	debugpy                   1.6.6           py311ha397e9f_0    conda-forge
	decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
	defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
	entrypoints               0.4                pyhd8ed1ab_0    conda-forge
	executing                 1.2.0              pyhd8ed1ab_0    conda-forge
	flit-core                 3.8.0              pyhd8ed1ab_0    conda-forge
	idna                      3.4                pyhd8ed1ab_0    conda-forge
	importlib-metadata        6.0.0              pyha770c72_0    conda-forge
	importlib_metadata        6.0.0                hd8ed1ab_0    conda-forge
	importlib_resources       5.10.2             pyhd8ed1ab_0    conda-forge
	ipykernel                 6.21.1             pyh736e0ef_0    conda-forge
	ipython                   8.10.0             pyhd1c38e8_0    conda-forge
	ipython_genutils          0.2.0                      py_1    conda-forge
	jedi                      0.18.2             pyhd8ed1ab_0    conda-forge
	jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
	json5                     0.9.5              pyh9f0ad1d_0    conda-forge
	jsonschema                4.17.3             pyhd8ed1ab_0    conda-forge
	jupyter_client            8.0.2              pyhd8ed1ab_0    conda-forge
	jupyter_core              5.2.0           py311h267d04e_0    conda-forge
	jupyter_events            0.5.0              pyhd8ed1ab_1    conda-forge
	jupyter_server            2.2.1              pyhd8ed1ab_0    conda-forge
	jupyter_server_fileid     0.6.0              pyhd8ed1ab_0    conda-forge
	jupyter_server_terminals  0.4.4              pyhd8ed1ab_1    conda-forge
	jupyter_server_ydoc       0.6.1              pyhd8ed1ab_0    conda-forge
	jupyter_ydoc              0.2.2              pyhd8ed1ab_0    conda-forge
	jupyterlab                3.6.1              pyhd8ed1ab_0    conda-forge
	jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
	jupyterlab_server         2.19.0             pyhd8ed1ab_0    conda-forge
	libcxx                    14.0.6               h2692d47_0    conda-forge
	libffi                    3.4.2                h3422bc3_5    conda-forge
	libsodium                 1.0.18               h27ca646_1    conda-forge
	libsqlite                 3.40.0               h76d750c_0    conda-forge
	libzlib                   1.2.13               h03a7124_4    conda-forge
	markupsafe                2.1.2           py311he2be06e_0    conda-forge
	matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
	mistune                   2.0.5              pyhd8ed1ab_0    conda-forge
	nbclassic                 0.5.1              pyhd8ed1ab_0    conda-forge
	nbclient                  0.7.2              pyhd8ed1ab_0    conda-forge
	nbconvert                 7.2.9              pyhd8ed1ab_0    conda-forge
	nbconvert-core            7.2.9              pyhd8ed1ab_0    conda-forge
	nbconvert-pandoc          7.2.9              pyhd8ed1ab_0    conda-forge
	nbformat                  5.7.3              pyhd8ed1ab_0    conda-forge
	ncurses                   6.3                  h07bb92c_1    conda-forge
	nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
	notebook                  6.5.2              pyha770c72_1    conda-forge
	notebook-shim             0.2.2              pyhd8ed1ab_0    conda-forge
	openssl                   3.0.8                h03a7124_0    conda-forge
	packaging                 23.0               pyhd8ed1ab_0    conda-forge
	pandoc                    2.19.2               hce30654_1    conda-forge
	pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
	parso                     0.8.3              pyhd8ed1ab_0    conda-forge
	pexpect                   4.8.0              pyh1a96a4e_2    conda-forge
	pickleshare               0.7.5                   py_1003    conda-forge
	pip                       23.0               pyhd8ed1ab_0    conda-forge
	pkgutil-resolve-name      1.3.10             pyhd8ed1ab_0    conda-forge
	platformdirs              3.0.0              pyhd8ed1ab_0    conda-forge
	prometheus_client         0.16.0             pyhd8ed1ab_0    conda-forge
	prompt-toolkit            3.0.36             pyha770c72_0    conda-forge
	psutil                    5.9.4           py311he2be06e_0    conda-forge
	ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
	pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
	pycparser                 2.21               pyhd8ed1ab_0    conda-forge
	pygments                  2.14.0             pyhd8ed1ab_0    conda-forge
	pyopenssl                 23.0.0             pyhd8ed1ab_0    conda-forge
	pyrsistent                0.19.3          py311he2be06e_0    conda-forge
	pysocks                   1.7.1              pyha2e5f31_6    conda-forge
	python                    3.11.0          h3ba56d0_1_cpython    conda-forge
	python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
	python-fastjsonschema     2.16.2             pyhd8ed1ab_0    conda-forge
	python-json-logger        2.0.4              pyhd8ed1ab_0    conda-forge
	python_abi                3.11                    3_cp311    conda-forge
	pytz                      2022.7.1           pyhd8ed1ab_0    conda-forge
	pyyaml                    6.0             py311he2be06e_5    conda-forge
	pyzmq                     25.0.0          py311h0f351f6_0    conda-forge
	readline                  8.1.2                h46ed386_0    conda-forge
	requests                  2.28.2             pyhd8ed1ab_0    conda-forge
	send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
	setuptools                67.1.0             pyhd8ed1ab_0    conda-forge
	six                       1.16.0             pyh6c4a22f_0    conda-forge
	sniffio                   1.3.0              pyhd8ed1ab_0    conda-forge
	soupsieve                 2.3.2.post1        pyhd8ed1ab_0    conda-forge
	stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
	terminado                 0.17.1             pyhd1c38e8_0    conda-forge
	tinycss2                  1.2.1              pyhd8ed1ab_0    conda-forge
	tk                        8.6.12               he1e0b03_0    conda-forge
	tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
	tornado                   6.2             py311he2be06e_1    conda-forge
	traitlets                 5.9.0              pyhd8ed1ab_0    conda-forge
	typing-extensions         4.4.0                hd8ed1ab_0    conda-forge
	typing_extensions         4.4.0              pyha770c72_0    conda-forge
	tzdata                    2022g                h191b570_0    conda-forge
	urllib3                   1.26.14            pyhd8ed1ab_0    conda-forge
	wcwidth                   0.2.6              pyhd8ed1ab_0    conda-forge
	webencodings              0.5.1                      py_1    conda-forge
	websocket-client          1.5.1              pyhd8ed1ab_0    conda-forge
	wheel                     0.38.4             pyhd8ed1ab_0    conda-forge
	xz                        5.2.6                h57fd34a_0    conda-forge
	y-py                      0.5.5           py311heb423af_2    conda-forge
	yaml                      0.2.5                h3422bc3_2    conda-forge
	ypy-websocket             0.8.2              pyhd8ed1ab_0    conda-forge
	zeromq                    4.3.4                hbdafb3b_1    conda-forge
	zipp                      3.13.0             pyhd8ed1ab_0    conda-forge

conda env:
	name: jlab-test
	channels:
	  - conda-forge
	dependencies:
	  - aiofiles=22.1.0=pyhd8ed1ab_0
	  - aiosqlite=0.18.0=pyhd8ed1ab_0
	  - anyio=3.6.2=pyhd8ed1ab_0
	  - appnope=0.1.3=pyhd8ed1ab_0
	  - argon2-cffi=21.3.0=pyhd8ed1ab_0
	  - argon2-cffi-bindings=21.2.0=py311he2be06e_3
	  - asttokens=2.2.1=pyhd8ed1ab_0
	  - attrs=22.2.0=pyh71513ae_0
	  - babel=2.11.0=pyhd8ed1ab_0
	  - backcall=0.2.0=pyh9f0ad1d_0
	  - backports=1.0=pyhd8ed1ab_3
	  - backports.functools_lru_cache=1.6.4=pyhd8ed1ab_0
	  - beautifulsoup4=4.11.2=pyha770c72_0
	  - bleach=6.0.0=pyhd8ed1ab_0
	  - brotlipy=0.7.0=py311he2be06e_1005
	  - bzip2=1.0.8=h3422bc3_4
	  - ca-certificates=2022.12.7=h4653dfc_0
	  - certifi=2022.12.7=pyhd8ed1ab_0
	  - cffi=1.15.1=py311hae827db_3
	  - charset-normalizer=2.1.1=pyhd8ed1ab_0
	  - comm=0.1.2=pyhd8ed1ab_0
	  - cryptography=39.0.1=py311h507f6e9_0
	  - debugpy=1.6.6=py311ha397e9f_0
	  - decorator=5.1.1=pyhd8ed1ab_0
	  - defusedxml=0.7.1=pyhd8ed1ab_0
	  - entrypoints=0.4=pyhd8ed1ab_0
	  - executing=1.2.0=pyhd8ed1ab_0
	  - flit-core=3.8.0=pyhd8ed1ab_0
	  - idna=3.4=pyhd8ed1ab_0
	  - importlib-metadata=6.0.0=pyha770c72_0
	  - importlib_metadata=6.0.0=hd8ed1ab_0
	  - importlib_resources=5.10.2=pyhd8ed1ab_0
	  - ipykernel=6.21.1=pyh736e0ef_0
	  - ipython=8.10.0=pyhd1c38e8_0
	  - ipython_genutils=0.2.0=py_1
	  - jedi=0.18.2=pyhd8ed1ab_0
	  - jinja2=3.1.2=pyhd8ed1ab_1
	  - json5=0.9.5=pyh9f0ad1d_0
	  - jsonschema=4.17.3=pyhd8ed1ab_0
	  - jupyter_client=8.0.2=pyhd8ed1ab_0
	  - jupyter_core=5.2.0=py311h267d04e_0
	  - jupyter_events=0.5.0=pyhd8ed1ab_1
	  - jupyter_server=2.2.1=pyhd8ed1ab_0
	  - jupyter_server_fileid=0.6.0=pyhd8ed1ab_0
	  - jupyter_server_terminals=0.4.4=pyhd8ed1ab_1
	  - jupyter_server_ydoc=0.6.1=pyhd8ed1ab_0
	  - jupyter_ydoc=0.2.2=pyhd8ed1ab_0
	  - jupyterlab=3.6.1=pyhd8ed1ab_0
	  - jupyterlab_pygments=0.2.2=pyhd8ed1ab_0
	  - jupyterlab_server=2.19.0=pyhd8ed1ab_0
	  - libcxx=14.0.6=h2692d47_0
	  - libffi=3.4.2=h3422bc3_5
	  - libsodium=1.0.18=h27ca646_1
	  - libsqlite=3.40.0=h76d750c_0
	  - libzlib=1.2.13=h03a7124_4
	  - markupsafe=2.1.2=py311he2be06e_0
	  - matplotlib-inline=0.1.6=pyhd8ed1ab_0
	  - mistune=2.0.5=pyhd8ed1ab_0
	  - nbclassic=0.5.1=pyhd8ed1ab_0
	  - nbclient=0.7.2=pyhd8ed1ab_0
	  - nbconvert=7.2.9=pyhd8ed1ab_0
	  - nbconvert-core=7.2.9=pyhd8ed1ab_0
	  - nbconvert-pandoc=7.2.9=pyhd8ed1ab_0
	  - nbformat=5.7.3=pyhd8ed1ab_0
	  - ncurses=6.3=h07bb92c_1
	  - nest-asyncio=1.5.6=pyhd8ed1ab_0
	  - notebook=6.5.2=pyha770c72_1
	  - notebook-shim=0.2.2=pyhd8ed1ab_0
	  - openssl=3.0.8=h03a7124_0
	  - packaging=23.0=pyhd8ed1ab_0
	  - pandoc=2.19.2=hce30654_1
	  - pandocfilters=1.5.0=pyhd8ed1ab_0
	  - parso=0.8.3=pyhd8ed1ab_0
	  - pexpect=4.8.0=pyh1a96a4e_2
	  - pickleshare=0.7.5=py_1003
	  - pip=23.0=pyhd8ed1ab_0
	  - pkgutil-resolve-name=1.3.10=pyhd8ed1ab_0
	  - platformdirs=3.0.0=pyhd8ed1ab_0
	  - prometheus_client=0.16.0=pyhd8ed1ab_0
	  - prompt-toolkit=3.0.36=pyha770c72_0
	  - psutil=5.9.4=py311he2be06e_0
	  - ptyprocess=0.7.0=pyhd3deb0d_0
	  - pure_eval=0.2.2=pyhd8ed1ab_0
	  - pycparser=2.21=pyhd8ed1ab_0
	  - pygments=2.14.0=pyhd8ed1ab_0
	  - pyopenssl=23.0.0=pyhd8ed1ab_0
	  - pyrsistent=0.19.3=py311he2be06e_0
	  - pysocks=1.7.1=pyha2e5f31_6
	  - python=3.11.0=h3ba56d0_1_cpython
	  - python-dateutil=2.8.2=pyhd8ed1ab_0
	  - python-fastjsonschema=2.16.2=pyhd8ed1ab_0
	  - python-json-logger=2.0.4=pyhd8ed1ab_0
	  - python_abi=3.11=3_cp311
	  - pytz=2022.7.1=pyhd8ed1ab_0
	  - pyyaml=6.0=py311he2be06e_5
	  - pyzmq=25.0.0=py311h0f351f6_0
	  - readline=8.1.2=h46ed386_0
	  - requests=2.28.2=pyhd8ed1ab_0
	  - send2trash=1.8.0=pyhd8ed1ab_0
	  - setuptools=67.1.0=pyhd8ed1ab_0
	  - six=1.16.0=pyh6c4a22f_0
	  - sniffio=1.3.0=pyhd8ed1ab_0
	  - soupsieve=2.3.2.post1=pyhd8ed1ab_0
	  - stack_data=0.6.2=pyhd8ed1ab_0
	  - terminado=0.17.1=pyhd1c38e8_0
	  - tinycss2=1.2.1=pyhd8ed1ab_0
	  - tk=8.6.12=he1e0b03_0
	  - tomli=2.0.1=pyhd8ed1ab_0
	  - tornado=6.2=py311he2be06e_1
	  - traitlets=5.9.0=pyhd8ed1ab_0
	  - typing-extensions=4.4.0=hd8ed1ab_0
	  - typing_extensions=4.4.0=pyha770c72_0
	  - tzdata=2022g=h191b570_0
	  - urllib3=1.26.14=pyhd8ed1ab_0
	  - wcwidth=0.2.6=pyhd8ed1ab_0
	  - webencodings=0.5.1=py_1
	  - websocket-client=1.5.1=pyhd8ed1ab_0
	  - wheel=0.38.4=pyhd8ed1ab_0
	  - xz=5.2.6=h57fd34a_0
	  - y-py=0.5.5=py311heb423af_2
	  - yaml=0.2.5=h3422bc3_2
	  - ypy-websocket=0.8.2=pyhd8ed1ab_0
	  - zeromq=4.3.4=hbdafb3b_1
	  - zipp=3.13.0=pyhd8ed1ab_0
	prefix: /Users/matthew/miniforge3/envs/jlab-test
</pre>
</details>

**Comment by krassowski** [comment]
This looks related to https://github.com/jupyter-server/jupyter_server/issues/1198.

**Comment by krassowski** [comment]
Interestingly I could not reproduce (not using conda so not exact same environment, but the differences could help pinpoint the issue). Does it also happen with the following:

```python
import multiprocessing
import foolib
del __file__
multiprocessing.Process(target=foolib.foo).start()
```

?

**Comment by mrahtz** [comment]
Thanks for the quick response!

Huh, interesting - this does indeed work fine:

```python
import multiprocessing
import foolib
del __file__
multiprocessing.Process(target=foolib.foo).start()
```

> This looks related to https://github.com/jupyter-server/jupyter_server/issues/1198.

Ah yeah, this looks to be exactly the same thing - downgrading my `ipykernel` to 6.20.2 fixes it too - and upgrading it to 6.21.0 breaks it again.

Should we close this to avoid the duplicate issue, or keep it open for visibility to other folks?

**Comment by krassowski** [comment]
Thank you for getting back. I will flag this as answered which will keep it open for a while and auto-close later, just so that anyone with the same problem can find the link to the issue in jupyter-server easilt (although the issue itself is in ipykernel, I guess we will continue discussing in jupyter-server for now).
